packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,111 @@
+import { Component, h, Prop } from '@stencil/core';
+import merge from 'lodash.merge';
+import { DataPoint, DataStream, Primitive, ViewPortConfig } from '../../../utils/dataTypes';
+import { isNumberDataStream } from '../../../utils/predicates';
+import { Threshold } from '../../charts/common/types';
+import { DialSizeConfig, RecursivePartial, DialMessages } from '../type';
+import { DefaultDialMessages } from '../util';
+import { DialLoading } from './sc-dial-loading';
+
+const title = (dataStream: { detailedName?: any; name?: any } | null | false | undefined) => {
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+
+const defaultUnit = '%';
+
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: DialSizeConfig;
+
+  @Prop() messageOverrides?: RecursivePartial<DialMessages>;
+
+  @Prop() significantDigits?: number;
+
+  @Prop() isLoading?: boolean = false;
+
+  private messages: DialMessages;
+  private unit: string;
+
+  componentWillLoad() {
+    this.messages = merge(DefaultDialMessages, this.messageOverrides);
+  }
+
+  render() {
+    const { yMin = 0, yMax = 0 } = this.viewport;
+    const propertyStream = this.propertyStream && isNumberDataStream(this.propertyStream) ? this.propertyStream : null;
+    const point = propertyStream ? this.propertyPoint : undefined;
+    const ifShowDefaultError = !propertyStream || (point && (point.y < yMin || point.y > yMax));
+    const error = this.propertyStream
+      ? this.propertyStream.error || (ifShowDefaultError && this.messages.error.dataNotNumberError)
+      : null;
+
+    const percent = point ? (point.y as number) / (yMax - yMin) : 0;
+    const unit = propertyStream && propertyStream.unit;
+    this.unit = propertyStream ? unit || defaultUnit : '';
```

Comment:
is there a reason to store this as a private class member?

Reply:
Done --- Change `unit` to `propertyStreamUnit` and change `this.unit`  to `unit`.